src/main/java/uk/gov/hmcts/reform/hmc/service/HearingManagementServiceImpl.java

```diff
@@ -440,4 +483,55 @@ private void sendRequestToQueue(HmiDeleteHearingRequest hmiDeleteHearingRequest,
         var jsonNode = objectMapperService.convertObjectToJsonNode(hmiDeleteHearingRequest);
         messageSenderToQueueConfiguration.sendMessageToQueue(jsonNode.toString(), hearingId, messageType);
     }
+
+    private void validateHearingActualsStatus(Long hearingId, String errorMessage) {
+        String status = getStatus(hearingId);
+        boolean isValidStatus = DeleteHearingStatus.isValidHearingActuals(status);
+        LocalDate minStartDate = hearingIdValidator.filterHearingResponses(hearingRepository.findById(hearingId).get());
+        LocalDate now = LocalDate.now();
+        if (isValidStatus && (minStartDate.isBefore(now) || minStartDate.equals(now))) {
+            throw new BadRequestException(errorMessage);
+        }
+    }
+
+    private void validateHearingOutcomeInformation(Long hearingId, String errorMessage) {
+        Optional<ActualHearingEntity> entity = getActualHearing(hearingId);
+        if (entity.isEmpty()) {
+            throw new BadRequestException(errorMessage);
+        }
+    }
+
+    private void validateHearingResultType(Long hearingId, String errorMessage) {
+        Optional<ActualHearingEntity> entity = getActualHearing(hearingId);
+        if (entity.isPresent()) {
+            HearingResultType hearingResultType = entity.get().getHearingResultType();
+
+            if ((hearingResultType.getLabel().equals(COMPLETED.getLabel())
+                || hearingResultType.getLabel().equals(ADJOURNED.getLabel()))
+                && actualHearingDayRepository.findByActualHearing(entity.get()).isEmpty()) {
+                throw new BadRequestException(errorMessage);
+            }
+        }
+    }
+
+    private Optional<ActualHearingEntity> getActualHearing(Long hearingId) {
+        Optional<HearingResponseEntity> hearingResponseEntity = hearingIdValidator
+            .getHearingResponse(hearingRepository.findById(hearingId).get());
+        if (hearingResponseEntity.isPresent()) {
+            return actualHearingRepository.findByHearingResponse(hearingResponseEntity.get());
```

Comment:
Maybe we could trow some nice exception. In case when consuming service will call this before creating hearing actuals.

Reply:
better to leave this, and leave the decision to throw the exception from caller based on the requirements